client/store/single-order.js

```diff
@@ -32,11 +34,24 @@ export const createGetOrderForAdminThunk = (id) => {
   }
 }
 
+export const createEditOrderForAdminThunk = (id, status) => {
+  return async (dispatch) => {
+    try{
+      await axios.put(`/api/admin/orders/${id}`, status)
+      dispatch(editOrderForAdminAction(status))
+    }catch(error) {
+      console.error(error)
+    }
+  }
+}
+
 // reducer
 export default (state = initialState, action) => {
   switch (action.type) {
     case GOT_ORDER:
       return action.order
+    case EDIT_ORDER_FOR_ADMIN:
+      return {...state, status: action.status.status}
```

Comment:
This goes back to how the event handler was written and how it was behaving.  Would you like to take this version so that you can move ahead, and I can refactor it later today?